content/blog/enforcing-and-automatically-tagging-aws-resources/index.md

```diff
@@ -0,0 +1,747 @@
+                    }
+                }
+            }
+        },
+    }],
+});
+
+interface AwsTagsPolicyConfig {
+    requiredTags?: string[];
+}
+```
+
+```go
+import * as policy from "@pulumi/policy";
+import { isTaggable } from "../lib/taggable";
+
+new policy.PolicyPack("aws-tags-policies", {
+    policies: [{
+        name: "check-required-tags",
+        description: "Ensure required tags are present on all AWS resources.",
+        configSchema: {
+            properties: {
+                requiredTags: {
+                    type: "array",
+                    items: { type: "string" },
+                },
+            },
+        },
+        validateResource: (args, reportViolation) => {
+            const config = args.getConfig<AwsTagsPolicyConfig>();
+            const requiredTags = config.requiredTags;
+            if (requiredTags && isTaggable(args.type)) {
+                const ts = args.props[tags];
+                for (const rt of requiredTags) {
+                    if (!ts || !ts[rt]) {
+                        reportViolation(
+                            `Taggable resource '${args.urn}' is missing required tag '${rt}'`);
+                    }
+                }
+            }
+        },
+    }],
+});
+
+interface AwsTagsPolicyConfig {
+    requiredTags?: string[];
+}
+```
+
+```csharp
+import * as policy from "@pulumi/policy";
+import { isTaggable } from "../lib/taggable";
+
+new policy.PolicyPack("aws-tags-policies", {
+    policies: [{
+        name: "check-required-tags",
+        description: "Ensure required tags are present on all AWS resources.",
+        configSchema: {
+            properties: {
+                requiredTags: {
+                    type: "array",
+                    items: { type: "string" },
+                },
+            },
+        },
+        validateResource: (args, reportViolation) => {
+            const config = args.getConfig<AwsTagsPolicyConfig>();
+            const requiredTags = config.requiredTags;
+            if (requiredTags && isTaggable(args.type)) {
+                const ts = args.props[tags];
+                for (const rt of requiredTags) {
+                    if (!ts || !ts[rt]) {
+                        reportViolation(
+                            `Taggable resource '${args.urn}' is missing required tag '${rt}'`);
+                    }
+                }
+            }
+        },
+    }],
+});
+
+interface AwsTagsPolicyConfig {
+    requiredTags?: string[];
+}
+```
+
+This project defines a _policy pack_ containing a set of _policy rules_, in this case, just one. That rule takes in a configurable set of tags to ensure exist on every taggable AWS resource. If a tag is missing, that is reported as a violation, and the deployment fails.
+
+> This leverages a library that helps to identify taggable AWS resources (see this gist for the full details). Although this policy is written in TypeScript, it can be applied to stacks written in any language.
```

Comment:
```suggestion
> This leverages a library that helps to identify taggable AWS resources (see this [gist](LINK MISSING?) for the full details). Although this policy is written in TypeScript, it can be applied to stacks written in any language.
```

Reply:
For sure, I'll be publishing the gist and linking to it before the post goes live.